frontend/src/components/Timer.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState, useEffect } from "react";
+
+interface Props {
+  phase: string;
+  phaseEndsAt: number;
+  round: number;
+}
+
+const Timer: React.FunctionComponent<Props> = ({
+  phase,
+  phaseEndsAt,
+  round,
+}: Props) => {
+  const initialTimerState = Math.round((phaseEndsAt - Date.now()) / 1000);
+  const [timeLeft, setTimeLeft] = useState(initialTimerState);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setTimeLeft(timeLeft => timeLeft - 1);
```

Comment:
This does not work :( useInterval is called only on first render, so the new phaseEndsAt values are ignored. You can update this value directly using the phaseEndsAt (you have to add phaseEndsAt to the list of dependencies, but that makes it even more reactive :))